### PR TITLE
Mongodb

### PR DESCRIPTION
**Pull request by Samet91**

### CODE REVIEW

File: src/app/Pages/Dashboard.tsx [hunk 1]
```diff
@@ -15,9 +15,21 @@ export default function Dashboard(): JSX.Element {
     []
   )
 
-  function handleNewTransaction(transaction: TransactionProps) {
+  async function handleNewTransaction(transaction: TransactionProps) {
     const newTransactions = [...transactions, transaction]
     setTransactions(newTransactions)
```
Comment (F-Kirchhoff) [inline src/app/Pages/Dashboard.tsx:16]: I would send only the new transaction to the database, and then refetch all transactions. then I guess you dont even need the state anymore.

File: src/server.ts [hunk 2]
```diff
@@ -2,17 +2,23 @@ import dotenv from 'dotenv'
...
-import { connectDatabase } from './app/hooks/database'
+import { connectDatabase, getCollection } from './app/hooks/database'
 
 if (!process.env.MONGODB_URI) {
   throw new Error('No MongoDB URL dotenv variable')
 }
 
 const app = express()
-const port = process.env.PORT || 3001
+const port = process.env.PORT || 3000
```
Comment (F-Kirchhoff) [inline src/server.ts:12]: I guess your frontend is served via 3000, so your backend has to be on 3001?

File: src/server.ts [hunk 3]
```diff
@@ -2,17 +2,23 @@ import dotenv from 'dotenv'
...
 
 app.use(express.json())
 
+app.post('/costs', async (req, res) => {
+  const newCosts = req.body
+  await getCollection().insertOne(newCosts)
+  res.status(200).send('Newcost was added')
+})
+
```
Comment (F-Kirchhoff) [inline src/server.ts:21]: nice!

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 2 above)
Comment (Samet91) [reply src/server.ts:12]: I think both must be on 3000?!

File: src/server.ts (hunk 3 above)
Comment (Samet91) [reply src/server.ts:21]: Thanks!

File: src/app/Pages/Dashboard.tsx [hunk 4]
```diff
@@ -15,9 +15,21 @@ export default function Dashboard(): JSX.Element {
...
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(transaction),
+    })
+    if (response.status === 200) {
+      console.log('added in database')
+    } else {
+      console.log('scheiße gelaufen')
```
Comment (janhendrikstruve) [inline src/app/Pages/Dashboard.tsx:27]: XD

File: src/server.ts (hunk 2 above)
Comment (janhendrikstruve) [reply src/server.ts:12]: i Think every service, like vite, storybook, express etc needs its own port like Felix said.

**Review by janhendrikstruve — COMMENTED** [review]
Very nice Code!

**Pull request merged**